### PR TITLE
[#175798933] Polling to check transaction status

### PR DESCRIPTION
**Pull request by infantesimone**
#### Description 

The goal of this PR is to add polling to check transaction status in response page for no 3ds flow.

#### List of Changes

- fixed api spec;
- added one api invocation to check if it is 3ds flow ;
- added polling to check transaction status;

#### Motivation and Context

This PR is important to handle transaction status in no 3ds flow.

#### How Has This Been Tested?

- yarn run test
- yarn start (with local [Payment Manager](https://github.com/pagopa/pm-mock))

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/response.ts [hunk 1]
```diff
@@ -27,10 +63,25 @@ document.addEventListener('DOMContentLoaded', async () => {
...
   }
+
+  await getDataFromSessionStorageTask('payment')
+    .chain(transaction => checkStatusTask(transaction.token, paymentManagerClient))
+    .chain(transactionStatusResponse => isNot3dsFlowTask(transactionStatusResponse))
+    .fold(
+      _ => showErrorStatus(), // 3ds case
+      transactionStatusResponse =>
+        checkStatusTask(
+          Buffer.from(transactionStatusResponse.data.idTransaction.toString()).toString('base64'),
```
Comment (GiovanniMancini) [inline src/response.ts:74]: @infantesimone isn't this the `token` field of `payment` data strucure in session storage?

File: openapi/api.yml [hunk 2]
```diff
@@ -2788,7 +2797,7 @@ definitions:
   WalletResponse:
     type: object
     required:
-      - data
+        - data
```
Comment (GiovanniMancini) [inline openapi/api.yml:2800]: @infantesimone Is this alignment required?

File: src/utils/transactionHelper.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+export const isNot3dsFlowTask = (
+  transactionStatusResponse: TransactionStatusResponse,
+): TaskEither<UNKNOWN, TransactionStatusResponse> =>
+  fromPredicate(
+    (transaction: TransactionStatusResponse) => fromNullable(transaction.data?.acsUrl).isNone(),
+    _ => UNKNOWN.value,
+  )(transactionStatusResponse);
+
+export const showErrorStatus = () => {
```
Comment (GiovanniMancini) [inline src/utils/transactionHelper.ts:41]: @infantesimone Maybe we should move rendering helpers to a different module

**Review by GiovanniMancini — APPROVED** [review]

File: src/response.ts (hunk 1 above)
Comment (infantesimone) [reply src/response.ts:74]: Yes it's the same, but it recomputed based on the transaction response, is it better to use the same?

File: openapi/api.yml (hunk 2 above)
Comment (infantesimone) [reply openapi/api.yml:2800]: thanks, that's a mistake.

File: src/utils/transactionHelper.ts (hunk 3 above)
Comment (infantesimone) [reply src/utils/transactionHelper.ts:41]: I agree, for now I can move it to response.pug?

Comment (pasqualespica) [inline src/utils/transactionHelper.ts:41]: Could be useful to pass `idStatus` to  `showErrorStatus ` too ?
So when we have a transaction status automa we could associate the correct message to show to the user. 
Also seen the great work present in the `TransactionStatesTypes` retrived from PM .

**Review by pasqualespica — APPROVED** [review]
Great work 🚀 
With this new polling fetch condition you have obtained a new _fp's belt_  👍

File: src/utils/transactionHelper.ts (hunk 3 above)
Comment (infantesimone) [reply src/utils/transactionHelper.ts:41]: Good point, when the messages to be shown are defined this part will have to be refactored.

**Pull request merged**